core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java

```diff
@@ -107,36 +108,103 @@ private void validateAndLoadDefaultProperties() {
     validateCarbonCSVReadBufferSizeByte();
     validateHandoffSize();
     validateCombineSmallInputFiles();
+    // The method validate the validity of configured carbon.timestamp.format value
+    // and reset to default value if validation fail
+    validateCarbonKey(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT,
+        CarbonCommonConstants.CARBON_TIMESTAMP_DEFAULT_FORMAT);
+    // The method validate the validity of configured carbon.date.format value
+    // and reset to default value if validation fail
+    validateCarbonKey(CarbonCommonConstants.CARBON_DATE_FORMAT,
```

Comment:
These validation should also be done when addProperty is called

Reply:
added validation check while addProperty also